src/What4/ExprHelpers.hs

```diff
@@ -294,8 +295,12 @@ rewriteSubExprs'' sym cache f e_outer = do
   -- This ensures that the What4 expression builder correctly re-establishes any invariants it requires
   -- after rebinding.
   Pair vars vals <- return $ toBindings binds
-  fn <- W4.definedFn sym W4.emptySymbol vars e' W4.AlwaysUnfold
-  W4.applySymFn sym fn vals >>= fixMux sym
+  case Ctx.viewSize (Ctx.size vals) of
+    -- no replacement
```

Comment:
It seems like this is a subtle change - can you document what would go wrong if you didn't special case this? I'm assuming this was our sneaky source of unsoundness

Reply:
It wasn't a source of unsoundness, but I suspected that it was trashing expression nonces in some cases and causing cache misses that would otherwise hit. I can drop the change from this PR if we want, but it shouldn't actually affect any semantics.

Reply:
No reason not to include it - I just like to understand subtle changes to avoid surprise.